### PR TITLE
Bloodsuckers objective tweaks and added epilogues

### PR DESCRIPTION
**Pull request by Chubbygummibear**
# Document the changes in your pull request

Since Bloodsucker mechanics kinda force them out of hiding by using blood as a timer it's actually a bit of an ask to survive the whole round without alerting people. i think this is why fulp has "make a lair" and "survive" as primary objectives and everything else as optional. i have implemented this **HOWEVER** i don't think you should get a big greentext for surviving, so you get a special marooned text that's more neutral if you do the bare minimum. 

Additionally everyone gets a little epilogue lore blurb in the credits screen based on how you did as a sucker. this takes into account if you did primary objectives, optional objectives, broke the masquerade or not, and even evac'd on the shuttle. Evacing is kinda a big risk due to the sol cycle so getting an evac ending is pretty impressive and hopefully your epilogue reflects that.

Epilogue text color is also dynamic based on how i gauged the outcomes. i also set it up to be easy to add more blurbs if people want to contribute by adding some messages to the list.

Tattax also asked me to reduce the objectives assigned on spawn from 2 to 1.

image
image
image

# Wiki Documentation

Bloodsucker assigned objectives down to 1
spawn objectives that aren't lair and survive are "optional"
objectives you get from joining a clan ARE NOT optional
to get the epic greentext you need to complete both primary and optional objectives. otherwise you get a minor neutral ending

# Changelog

:cl:  @tattax, Truzkawa, @Skrem7, Chubbygummibear
rscadd: added code that supports optional objectives
rscadd: bloodsucker round-end report comes with epilogue blurbs based on how you did
tweak: bloodsucker assigned objectives at spawn are "optional" 
tweak: objectives you get for joining a clan through ranking up are NOT optional
tweak: epic greentext is only achieved through finishing both primary and optional. only primary will net you a neutral ending
/:cl:

### CODE REVIEW

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 1]
```diff
@@ -298,13 +303,79 @@
...
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && !escaped)
+		//finish primary objectives only, don't break masquerade, don't escape
+		flavor_message += pick(list(
+			"A low profile has always suited you best. Just enough to get by, don't make waves, and keep your head low. It's not luxorious living, but it beats being dead."
+		))
+		message_color = "#517fff"
+	else
+		//perish or just fuck up and fail your primary objectives
+		flavor_message += pick(list(
+			"Thus ends the story of [ReturnFullName(TRUE)]. Hopefully future generations will look back on your legacy and reflect on the lessons of the past."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"Thus ends the story of [ReturnFullName(TRUE)]. No doubt future generations will look back on your legacy and reflect on the lessons of the past. If they remember you at all."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 2]
```diff
@@ -298,13 +303,79 @@
...
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && !escaped)
+		//finish primary objectives only, break masquerade, don't escape
+		flavor_message += pick(list(
+			"You survived, but you broke the masquerade, and both monster hunters and bloodsuckers are going to know where to find you... Perhaps it's time to see what's so about special this Sol the cattle keep talking about."
+		))
+		message_color = "#ef2f3c"
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && escaped)
+		//finish primary objectives only, don't break masquerade, escape
+		flavor_message += pick(list(
+			"That station didn't suit you anyway, and so now it's time to find a new locale. Maybe your next haunt will prove more comfortable."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"A low profile has always suited you best, conspiring enough to satiate the clan and keep your head low. It's not luxorious living, though death is a less kind alternative. On to the next station."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 3]
```diff
@@ -298,13 +303,79 @@
...
+		))
+		message_color = "#517fff"
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && !escaped)
+		//finish primary objectives only, don't break masquerade, don't escape
+		flavor_message += pick(list(
+			"A low profile has always suited you best. Just enough to get by, don't make waves, and keep your head low. It's not luxorious living, but it beats being dead."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"You completed your mission and kept your identity free of heresy, though your mark here is not strong enough to lay a claim. Best stow away when the next shuttle comes around."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 4]
```diff
@@ -298,13 +303,79 @@
...
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && escaped)
+		//finish primary objectives only, break masquerade, escape
+		flavor_message += pick(list(
+			"You may have completed what was asked of you, but that doesn't change the fact that you violated the masqerade. Good thing you got off the station, because it's time to disappear."
+		))
+		message_color = "#517fff"
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && !escaped)
+		//finish primary objectives only, break masquerade, don't escape
+		flavor_message += pick(list(
+			"You survived, but you broke the masquerade, and both monster hunters and bloodsuckers are going to know where to find you... Perhaps it's time to see what's so about special this Sol the cattle keep talking about."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"You survived, but you broke the Masquerade, your blood-stained presence clear and your power limited. No doubt death in the form of claw or stake hails its approach. Perhaps it's time to understand the cattles' fascinations with the suns."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 5]
```diff
@@ -298,13 +303,79 @@
...
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && !escaped)
+		//finish all objectives, don't break masquerade, don't escape
+		flavor_message += pick(list(
+			"This station is your own slice of paradise. Each batch of crew is ripe for the picking of vassals and food. At this rate you might even start your own clan here."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && escaped)
+		//finish primary objectives only, break masquerade, escape
+		flavor_message += pick(list(
+			"You may have completed what was asked of you, but that doesn't change the fact that you violated the masqerade. Good thing you got off the station, because it's time to disappear."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"Your mission accomplished, you step off the spacecraft, feeling the mark of exile on your neck. Your allies gone, your veins thrum with a singular purpose: survival."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 6]
```diff
@@ -298,13 +303,79 @@
...
+			"You ran amok of the station, completed every task set before you, and killed any who opposed you. The Camarilla dare not risk a move against you for breaking the Masquerade."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && escaped)
+		//finish all objectives, don't break masquerade, escape
+		flavor_message += pick(list(
+			"You've proven your skills, you even managed to escape the station. Upon arriving back at Centom, an unassuming assistant hands you an invitation stamped with the Camerilla seal. High society is waiting for you."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && !escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:341]:
```suggestion
	else if(objectives_complete && optional_objectives_complete && !broke_masquerade || !escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 7]
```diff
@@ -298,13 +303,79 @@
...
+		//finish all objectives, don't break masquerade, don't escape
+		flavor_message += pick(list(
+			"This station is your own slice of paradise. Each batch of crew is ripe for the picking of vassals and food. At this rate you might even start your own clan here."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"This station has become your own slice of paradise. Your mission completed, you turn on the others who were stranded, ripe for your purposes. Who knows? If they prove to elevate your power enough, perhaps a new clan might be founded here."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 8]
```diff
@@ -298,13 +303,79 @@
...
+			"Describing you as a tyrant is a disservice to your abilities. You're a god amongst Bloodsuckers, and a demon in the eyes of everyone else. That station was just the beginning for you."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && broke_masquerade && !escaped)
+		//finish all objectives, break masquerade, don't evac
+		flavor_message += pick(list(
+			"You ran amok of the station, completed every task set before you, and killed any who opposed you. The Camarilla dare not risk a move against you for breaking the Masquerade."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:335]:
```suggestion
	else if(objectives_complete && optional_objectives_complete && !broke_masquerade || escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 9]
```diff
@@ -298,13 +303,79 @@
...
+			"You've proven your skills, you even managed to escape the station. Upon arriving back at Centom, an unassuming assistant hands you an invitation stamped with the Camerilla seal. High society is waiting for you."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && !escaped)
+		//finish all objectives, don't break masquerade, don't escape
+		flavor_message += pick(list(
+			"This station is your own slice of paradise. Each batch of crew is ripe for the picking of vassals and food. At this rate you might even start your own clan here."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:347]:
```suggestion
	else if(objectives_complete && !optional_objectives_complete || broke_masquerade && escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 10]
```diff
@@ -298,13 +303,79 @@
...
+		//finish primary objectives only, break masquerade, escape
+		flavor_message += pick(list(
+			"You may have completed what was asked of you, but that doesn't change the fact that you violated the masqerade. Good thing you got off the station, because it's time to disappear."
+		))
+		message_color = "#517fff"
+	else if(objectives_complete && !optional_objectives_complete && broke_masquerade && !escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:353]:
```suggestion
	else if(objectives_complete && !optional_objectives_complete || broke_masquerade && !escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 11]
```diff
@@ -298,13 +303,79 @@
...
+		//finish primary objectives only, break masquerade, don't escape
+		flavor_message += pick(list(
+			"You survived, but you broke the masquerade, and both monster hunters and bloodsuckers are going to know where to find you... Perhaps it's time to see what's so about special this Sol the cattle keep talking about."
+		))
+		message_color = "#ef2f3c"
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:359]:
```suggestion
	else if(objectives_complete && !optional_objectives_complete || !broke_masquerade || escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 12]
```diff
@@ -298,13 +303,79 @@
...
+	else if(objectives_complete && optional_objectives_complete && broke_masquerade && !escaped)
+		//finish all objectives, break masquerade, don't evac
+		flavor_message += pick(list(
+			"You ran amok of the station, completed every task set before you, and killed any who opposed you. The Camarilla dare not risk a move against you for breaking the Masquerade."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && !broke_masquerade && escaped)
+		//finish all objectives, don't break masquerade, escape
+		flavor_message += pick(list(
+			"You've proven your skills, you even managed to escape the station. Upon arriving back at Centom, an unassuming assistant hands you an invitation stamped with the Camerilla seal. High society is waiting for you."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"You step off the spacecraft with a mark of pride at a superbly completed mission. Upon arriving back at CentCom, an unassuming assistant palms you an invitation stamped with the Camarilla seal. High society awaits: a delicacy you have earned."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 13]
```diff
@@ -298,13 +303,79 @@
...
+			"You survived, but you broke the masquerade, and both monster hunters and bloodsuckers are going to know where to find you... Perhaps it's time to see what's so about special this Sol the cattle keep talking about."
+		))
+		message_color = "#ef2f3c"
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && escaped)
+		//finish primary objectives only, don't break masquerade, escape
+		flavor_message += pick(list(
+			"That station didn't suit you anyway, and so now it's time to find a new locale. Maybe your next haunt will prove more comfortable."
+		))
+		message_color = "#517fff"
+	else if(objectives_complete && !optional_objectives_complete && !broke_masquerade && !escaped)
```
Comment (tattax) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm:365]:
```suggestion
	else if(objectives_complete && !optional_objectives_complete || !broke_masquerade || !escaped)
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 14]
```diff
@@ -298,13 +303,79 @@
...
+	if(objectives_complete && optional_objectives_complete && broke_masquerade && escaped)
+		//finish all objectives, break masquerade, evac
+		flavor_message += pick(list(
+			"Describing you as a tyrant is a disservice to your abilities. You're a god amongst Bloodsuckers, and a demon in the eyes of everyone else. That station was just the beginning for you."
+		))
+		message_color = "#008000"
+	else if(objectives_complete && optional_objectives_complete && broke_masquerade && !escaped)
+		//finish all objectives, break masquerade, don't evac
+		flavor_message += pick(list(
+			"You ran amok of the station, completed every task set before you, and killed any who opposed you. The Camarilla dare not risk a move against you for breaking the Masquerade."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"Blood still pumps in your veins as you lay stranded on the station. No doubt the wake of chaos left in your path will attract danger, but greater power than you've ever felt courses through your body. Let the Camarilla and the witchers come. You will be waiting."
```

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 15]
```diff
@@ -298,13 +303,79 @@
...
+	var/list/flavor = list()
+	var/flavor_message
+	var/escaped = (owner.current.onCentCom() || owner.current.onSyndieBase())
+	flavor += "<div><font color='#6d6dff'>Epilogue: </font>"
+	var/message_color = "#ef2f3c"
+	//i used pick() in case anyone wants to add more messages as time goes on
+	if(objectives_complete && optional_objectives_complete && broke_masquerade && escaped)
+		//finish all objectives, break masquerade, evac
+		flavor_message += pick(list(
+			"Describing you as a tyrant is a disservice to your abilities. You're a god amongst Bloodsuckers, and a demon in the eyes of everyone else. That station was just the beginning for you."
```
Comment (Skrem7) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]:
```suggestion
			"What matters of the Masquerade to you? Let it crumble into dust as your tyranny whips forward to dine on more stations. News of your butchering exploits will quickly spread, and you know what will encompass the minds of mortals and undead alike. Fear."
```

**Review by Skrem7 — COMMENTED** [review]
Hi I'm a writer I like to write things

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm [hunk 16]
```diff
@@ -298,13 +303,79 @@
...
+		report += span_greentext(span_big("The [name] was successful!"))
+	else if(objectives_complete && !optional_objectives_complete)
+		report += span_marooned("The [name] survived, but has not made a name for [owner.current.p_them()]self...")
+	else		
+		report += span_redtext(span_big("The [name] has failed!"))
+	report += get_Flavor(objectives_complete, optional_objectives_complete)
 	return report
 
+/// Evaluates the conditions of the bloodsucker at the end of each round to pick a flavor message to add
+/datum/antagonist/bloodsucker/proc/get_Flavor(objectives_complete, optional_objectives_complete)
```
Comment (ToasterBiome) [inline code/modules/antagonists/bloodsuckers/bloodsuckers.dm]: This is the worst casing I have ever seen

**Review by ToasterBiome — DISMISSED** [review]

File: code/modules/antagonists/bloodsuckers/bloodsuckers.dm (hunk 16 above)
Comment (Chubbygummibear) [reply code/modules/antagonists/bloodsuckers/bloodsuckers.dm]: there

**Review by JamieD1 — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Comment by Chubbygummibear** [comment]
this needs to wait to be merged until #14648 is fixed or reverted, unless you want me to add a fix to this pr

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**